mail_debrand/models/mail_render_mixin.py

```diff
@@ -12,43 +12,61 @@
-                parent = elem.getparent()
-                previous = elem.getprevious()
+            # We don't want to change what was explicitly added in the message body,
+            # so we will only change what is before and after it.
+            if to_keep:
+                to_change = value.split(to_keep)
+            else:
+                to_change = [value]
+                to_keep = ""
+            new_parts = []
+            for part in to_change:
+                tree = etree.HTML(
+                    part
+                )  # html with broken links   tree = etree.fromstring(value) just xml
+                if tree is None:
+                    new_parts.append(part)
+                    continue
+                odoo_achors = tree.xpath('//a[contains(@href,"odoo.com")]')
+                for elem in odoo_achors:
+                    parent = elem.getparent()
+                    previous = elem.getprevious()
 
-                if remove_before and not remove_parent and previous is not None:
-                    # remove 'using' that is before <a and after </span>
-                    bytes_text = etree.tostring(
-                        previous, pretty_print=True, method="html"
-                    )
-                    only_what_is_in_tags = bytes_text[: bytes_text.rfind(b">") + 1]
-                    data_formatted = html.fromstring(only_what_is_in_tags)
-                    parent.replace(previous, data_formatted)
-                if remove_parent and len(parent.getparent()):
-                    # anchor <a href odoo has a parent powered by that must be removed
-                    parent.getparent().remove(parent)
-                else:
-                    if parent.tag == "td":  # also here can be powerd by
+                    if remove_before and not remove_parent and previous is not None:
+                        # remove 'using' that is before <a and after </span>
+                        bytes_text = etree.tostring(
+                            previous, pretty_print=True, method="html"
+                        )
+                        only_what_is_in_tags = bytes_text[: bytes_text.rfind(b">") + 1]
+                        data_formatted = html.fromstring(only_what_is_in_tags)
+                        parent.replace(previous, data_formatted)
+                    if remove_parent and len(parent.getparent()):
+                        # anchor <a href odoo has a parent powered by that must be removed
                         parent.getparent().remove(parent)
                     else:
-                        parent.remove(elem)
-            value = etree.tostring(tree, pretty_print=True, method="html")
-            # etree can return bytes; ensure we get a proper string
-            if type(value) is bytes:
-                value = value.decode()
-        return re.sub("[^(<)(</)]odoo", "", value, flags=re.IGNORECASE)
+                        if parent.tag == "td":  # also here can be powerd by
```

Comment:
I find it weird that you remove the parent even if you got `remove_parent=False`. Does that mean that we can just remove that argument from the method?

```suggestion
                        if parent.tag == "td":  # also here can be powered by
```